crates/vim/src/visual.rs

```diff
@@ -500,8 +500,30 @@ pub fn select_match(
     direction: Direction,
     cx: &mut WindowContext,
 ) {
-    let count = vim.take_count(cx).unwrap_or(1);
     let pane = workspace.active_pane().clone();
+    let mut exists_match = false;
+    pane.update(cx, |pane, cx| {
+        if let Some(search_bar) = pane.toolbar().read(cx).item_of_type::<BufferSearchBar>() {
+            search_bar.update(cx, |search_bar, cx| {
+                exists_match = search_bar.exists_match(cx);
+            });
+        }
+    });
+    
+    if !exists_match {
+        if !vim.workspace_state.replaying {
+            vim.clear_operator(cx);
+        } else {
+            move_cursor(vim, Motion::Right, Some(1), cx);
+        }
+        // clear recording
+        vim.workspace_state.recording = false;
+        vim.workspace_state.recorded_actions = Default::default();
+        vim.workspace_state.recorded_count = None;
+        return;
+    }
```

Comment:
@joaquin30 I think this check should be at the point we pop the operator. If there is no active match, then we just do nothing.